packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -70,78 +70,135 @@ export interface DeploymentCircuitBreaker {
   /**
    * HTTPCode_Target_2XX_Count Metric
+   * This metric is only available if you have configured Amazon ECS Service Connect and
+   * the appProtocol is HTTP or HTTP2 in the port mapping in the task definition.
+   * Units of this metric is Count
    */
-  HTTP_CODE_TARGET_2XX_COUNT = 'HTTPCode_Target_2XX_Count',
+  public static readonly HTTP_CODE_TARGET_2XX_COUNT = new EcsMetric('HTTPCode_Target_2XX_Count');
   /**
    * HTTPCode_Target_3XX_Count Metric
+   * This metric is only available if you have configured Amazon ECS Service Connect and
+   * the appProtocol is HTTP or HTTP2 in the port mapping in the task definition.
+   * Units of this metric is Count
    */
-  HTTP_CODE_TARGET_3XX_COUNT = 'HTTPCode_Target_3XX_Count',
+  public static readonly HTTP_CODE_TARGET_3XX_COUNT = new EcsMetric('HTTPCode_Target_3XX_Count');
   /**
    * HTTPCode_Target_4XX_Count Metric
+   * This metric is only available if you have configured Amazon ECS Service Connect and
+   * the appProtocol is HTTP or HTTP2 in the port mapping in the task definition.
+   * Units of this metric is Count
    */
-  HTTP_CODE_TARGET_4XX_COUNT = 'HTTPCode_Target_4XX_Count',
+  public static readonly HTTP_CODE_TARGET_4XX_COUNT = new EcsMetric('HTTPCode_Target_4XX_Count');
   /**
    * HTTPCode_Target_5XX_Count Metric
+   * This metric is only available if you have configured Amazon ECS Service Connect and
+   * the appProtocol is HTTP or HTTP2 in the port mapping in the task definition.
+   * Units of this metric is Count
    */
-  HTTP_CODE_TARGET_5XX_COUNT = 'HTTPCode_Target_5XX_Count',
+  public static readonly HTTP_CODE_TARGET_5XX_COUNT = new EcsMetric('HTTPCode_Target_5XX_Count');
   /**
    * RequestCountPerTarget Metric
+   * This metric is only available if you have configured Amazon ECS Service Connect.
+   * Units of this metric is Count
    */
-  REQUEST_COUNT_PER_TARGET = 'RequestCountPerTarget',
+  public static readonly REQUEST_COUNT_PER_TARGET = new EcsMetric('RequestCountPerTarget');
   /**
    * TargetProcessedBytes Metric
+   * This metric is only available if you have configured Amazon ECS Service Connect.
+   * Units of this metric is Count
    */
-  TARGET_PROCESSED_BYTES = 'TargetProcessedBytes',
+  public static readonly TARGET_PROCESSED_BYTES = new EcsMetric('TargetProcessedBytes');
   /**
    * TargetResponseTime Metric
+   * This metric is only available if you have configured Amazon ECS Service Connect.
+   * Units of this metric is Milliseconds
    */
-  TARGET_REESPONSE_TIME = 'TargetResponseTime',
+  public static readonly TARGET_RESPONSE_TIME = new EcsMetric('TargetResponseTime');
+
+  /**
+   * Add custom metric
+   */
+  public static custom(value: string) {
+    return new EcsMetric(value);
+  }
+
+  public constructor(
+    /**
+     * customValue refers to metric name
+     */
+    public readonly customValue: string,
+  ) { }
+
+  /**
+   * Determine if it's a service connect metric
+   */
+  public isServiceConnectMetric(): boolean {
```

Comment:
this seems quite fragile if we have to add new metrics; is there any way we can modify the constructor to store additional data for service connect metrics?